src/lib/consensus/proof_of_stake.ml

```diff
@@ -218,17 +216,25 @@ module Make (Inputs : Inputs_intf) :
       let in_seed_update_range_var (slot : Unpacked.var) =
```

Comment:
It would be nice to model the code a little more similarly to the original ocaml implementation. For example, splitting off `after_lock_checkpoint` here would have no impact on the construction of the snark dsl and would increase code comprehension, in my opinion.